libs/ts-rest/core/src/lib/client.ts

```diff
@@ -96,13 +107,16 @@ type DataReturnArgs<
   Without<DataReturnArgsBase<TRoute, TClientArgs>, never>
 >;
 
-export type ApiRouteResponse<T> =
+export type ApiRouteResponseStrict<T> =
```

Comment:
Can you make a changeset explaining this change? Since this is a behavior change (probably welcomed), it'll be helpful for users to know that these response types are now strict. 

Also, maybe there's a way we can provide a "loose" type? Not sure if people would want that, though.

Reply:
Will do, wanted to get some feedback on the direction first to see if this was of interest.